common/actions/wallet/actionCreators.ts

```diff
@@ -42,11 +42,27 @@ export function setWallet(value: IWallet): types.SetWalletAction {
   };
 }
 
-export type TSetBalance = typeof setBalance;
-export function setBalance(value: Wei): types.SetBalanceAction {
+export function setBalancePending(): types.SetBalancePendingAction {
   return {
-    type: constants.WALLET_SET_BALANCE,
-    payload: value
+    type: constants.WALLET_SET_BALANCE_PENDING,
+    payload: { status: { isPending: true } }
+  };
+}
+
+export type TSetBalance = typeof setBalanceFullfilled;
+export function setBalanceFullfilled(
+  value: Wei
+): types.SetBalanceFullfilledAction {
+  return {
+    type: constants.WALLET_SET_BALANCE_FULFILLED,
+    payload: { value, status: { isPending: false } }
+  };
+}
+
+export function SetBalanceRejected(): types.SetBalanceRejectedAction {
+  return {
+    type: constants.WALLET_SET_BALANCE_REJECTED,
+    payload: { status: { isPending: false } }
```

Comment:
Having a payload is unnecessary for rejected and pending, and isPending is unneeded for fulfilled. Redux's docs recommend:

> It's a good idea to pass as little data in each action as possible.

Because how the action affects the redux store should be handled in the reducer. Rather than passing a bunch of stuff, and doing a bunch of spreads, you should just set isPending to true or false inside of the reducer cases. This makes it easier to trace where that assignment actually happens, and allows changes to be simpler since we only need to change the reducer, not the actions.

Actions are more of your user's intent, while the reducer is how the code responds to that intent.

Reply:
I'll tidy that up. Passing as little data as possible in the payload is definitely preferable.